src/core/server/licensing/licensing_service.ts

```diff
@@ -0,0 +1,160 @@
+  private poller$!: Observable<number>;
+  private pollerSubscription!: Subscription;
+  private service$!: LicensingServiceSubject;
+  private license: any;
+
+  constructor(private readonly coreContext: CoreContext) {
+    this.logger = coreContext.logger.get(SERVICE_NAME);
+    this.config$ = coreContext.configService
+      .atPath<LicensingConfigType>(SERVICE_NAME)
+      .pipe(map(rawConfig => new LicensingConfig(rawConfig, coreContext.env)));
+  }
+
+  private hasLicenseInfoChanged(newLicense: any) {
+    return (
+      newLicense.mode !== this.license.mode ||
+      newLicense.status !== this.license.status ||
+      newLicense.expiry_date_in_millis !== this.license.expiry_date_in_millis
+    );
+  }
+
+  private async fetchInfo(
+    { http }: LicensingSetupDependencies,
+    clusterSource: string,
+    pollingFrequency: number
+  ) {
+    this.logger.debug(
+      `Calling [${clusterSource}] Elasticsearch _xpack API. Polling frequency: ${pollingFrequency}`
+    );
+
+    const cluster = http.server.plugins.elasticsearch.getCluster(clusterSource);
+
+    try {
+      const response = await cluster.callWithInternalUser('transport.request', {
+        method: 'GET',
+        path: '/_xpack',
+      });
+      const newLicense = (response && response.license) || {};
+      const features = (response && response.features) || {};
+      const licenseInfoChanged = this.hasLicenseInfoChanged(newLicense);
+
+      if (licenseInfoChanged) {
+        const licenseInfo = [
+          `mode: ${newLicense.mode}`,
+          `status: ${newLicense.status}`,
+          'expiry_date_in_millis' in newLicense &&
+            `expiry date: ${moment(newLicense.expiry_date_in_millis, 'x').format()}`,
+        ]
+          .filter(Boolean)
+          .join(' | ');
+
+        this.logger.info(
+          `Imported ${this.license ? 'changed ' : ''}license information` +
+            ` from Elasticsearch for the [${clusterSource}] cluster: ${licenseInfo}`
+        );
+
+        return { license: false, error: null, features };
+      }
+
+      return { license: newLicense, error: null, features };
+    } catch (err) {
+      this.logger.warn(
+        `License information could not be obtained from Elasticsearch` +
+          ` for the [${clusterSource}] cluster. ${err}`
+      );
+
+      return { license: null, error: err, features: {} };
+    }
+  }
+
+  private create(
+    { clusterSource, pollingFrequency }: LicensingConfig,
+    deps: LicensingSetupDependencies
+  ) {
+    if (this.service$) {
+      return this.service$;
+    }
+
+    const service$ = new BehaviorSubject<LicensingServiceSetup | null>(null);
+
+    this.poller$ = interval(pollingFrequency);
+    this.poller$.pipe(
+      switchMap(_ =>
+        from(this.fetchInfo(deps, clusterSource, pollingFrequency)).pipe(
+          tap(({ license, error, features }) => {
+            // If license is false, the license did not change and we don't need to push
+            // a new one
```

Comment:
On line 91 it returns false if the license _did_ change?

Reply:
Good catch!